expr/expr.go

```diff
@@ -781,44 +784,94 @@ func EvalExpr(e *expr, from, until int32, values map[MetricRequest][]*MetricData
 				}
 				return total[0].Values[i]
 			}
-			var totalString string
 			if e.args[1].etype == etName {
 				totalString = e.args[1].target
 			} else {
 				totalString = fmt.Sprintf("%s(%s)", e.args[1].target, e.args[1].argString)
 			}
-			formatName = func(a *MetricData) string {
-				return fmt.Sprintf("asPercent(%s,%s)", a.Name, totalString)
+			formatName = func(a, b string) string {
+				return fmt.Sprintf("asPercent(%s,%s)", a, b)
+			}
+		} else if len(e.args)%2 == 0 && (e.args[1].etype == etName || e.args[1].etype == etFunc) {
+			total, err := getSeriesArg(e.args[1], from, until, values)
+			if err != nil {
+				return nil, err
+			}
+			if len(total) != len(arg) {
+				return nil, ErrWildcardNotAllowed
+			}
+			multipleSeries = true
+			formatName = func(a, b string) string {
+				return fmt.Sprintf("asPercent(%s,%s)", a, b)
 			}
 		} else {
+			fmt.Printf("%v %v\n", len(e.args), len(arg))
 			return nil, errors.New("total must be either a constant or a series")
 		}
 
 		var results []*MetricData
 
-		for _, a := range arg {
-			r := *a
-			r.Name = formatName(a)
-			r.Values = make([]float64, len(a.Values))
-			r.IsAbsent = make([]bool, len(a.Values))
-			results = append(results, &r)
-		}
+		if multipleSeries {
+			for i := range e.args {
+				if i >= len(e.args)/2 {
+					break
+				}
+				arg1, err := getSeriesArg(e.args[i], from, until, values)
+				if err != nil {
+					return nil, err
+				}
+				arg2, err := getSeriesArg(e.args[len(e.args)/2+i], from, until, values)
+				if err != nil {
+					return nil, err
+				}
 
-		for i := range results[0].Values {
+				if len(arg1) != len(arg2) {
+					return nil, errors.New("Length mismatch, globs must return same amount of data")
+				}
+				for j := range arg1 {
+					a := arg1[j]
+					b := arg2[j]
+
+					r := *a
+					r.Name = formatName(a.Name, b.Name)
+					r.Values = make([]float64, len(a.Values))
+					r.IsAbsent = make([]bool, len(a.Values))
+					for k := range a.Values {
+						if a.IsAbsent[k] || b.IsAbsent[k] {
+							r.Values[k] = 0
+							r.IsAbsent[k] = true
+							continue
+						}
+						r.Values[k] = (a.Values[k] / b.Values[k]) * 100
+					}
+					results = append(results, &r)
+				}
+			}
+		} else {
+			for _, a := range arg {
+				r := *a
+				r.Name = formatName(a.Name, totalString)
+				r.Values = make([]float64, len(a.Values))
+				r.IsAbsent = make([]bool, len(a.Values))
+				results = append(results, &r)
+			}
 
-			total := getTotal(i)
+			for i := range results[0].Values {
 
-			for j := range results {
-				r := results[j]
-				a := arg[j]
+				total := getTotal(i)
 
-				if a.IsAbsent[i] || math.IsNaN(total) || total == 0 {
-					r.Values[i] = 0
-					r.IsAbsent[i] = true
-					continue
-				}
+				for j := range results {
```

Comment:
I've decided to leave that as is for 2 reasons:
1. If I modify stuff like that I still need to create a new "r" variable. Otherwise I got an error "source data was modified" during tests.
2. It's done in the same way as for "b" which makes it easier to read.